src/io/ctrees_utils.c

```diff
@@ -660,24 +672,24 @@ void assign_mergertree_indices(const int64_t totnhalos, struct halo_data *forest
 #undef PID_COMPARATOR
 #undef ID_COMPARATOR
 
-    return;
+    return EXIT_SUCCESS;
 }
 
 
 int64_t find_fof_halo(const int64_t totnhalos, const struct additional_info *info, const int start_loc, const int64_t upid, int verbose, int64_t calldepth)
 {
-    XASSERT(totnhalos < INT_MAX, EXIT_FAILURE,
+    XRETURN(totnhalos < INT_MAX, -EXIT_FAILURE,
             "Totnhalos must be less than %d. Otherwise indexing with int (start_loc) will break\n", INT_MAX);
     int64_t loc = -1;
     if(info[start_loc].pid == -1) {
         return start_loc;/* should never be called to find the FOF of a FOF halo -> but whatever*/
     }
 
-    const int64_t max_recursion_depth = 30, recursion_depth_for_verbose = 3;
+    const int64_t max_recursion_depth = 30, recursion_depth_for_verbose = 5;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'max_recursion_depth' is assigned a value that is never used.](https://app.codacy.com/app/jacobseiler/sage-model/pullRequest?prid=4171404)